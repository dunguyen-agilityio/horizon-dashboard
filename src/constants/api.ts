export const API_ENTITY = {
  COMPLEX_REPORT: 'complex-report',
  CHECK_REPORT: 'check-report',
  FINANCIAL_REPORT: 'financial-report',
  USERS: 'users',
  SIGN_IN: 'auth/local',
  SIGN_UP: 'auth/local/register',
  FORGET_PASSWORD: 'auth/forgot-password',
  RESET_PASSWORD: 'auth/reset-password',
  CONTRIBUTOR: 'contributors',
};
