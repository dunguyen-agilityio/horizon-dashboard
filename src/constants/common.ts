export const QUERY_KEY = 'q';
export const STEP_KEY = 'step';
