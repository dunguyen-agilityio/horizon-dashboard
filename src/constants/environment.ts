export const API_ENDPOINT = process.env.API_ENDPOINT!;

// Firebase
export const FIREBASE_API_KEY = process.env.NEXT_PUBLIC_FIREBASE_API_KEY;
export const FIREBASE_AUTH_DOMAIN =
  process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN;
export const FIREBASE_PROJECT_ID = process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID;
export const FIREBASE_STORAGE_BUCKET =
  process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET;
export const FIREBASE_SENDER_ID = process.env.NEXT_PUBLIC_FIREBASE_SENDER_ID;
export const FIREBASE_APP_ID = process.env.NEXT_PUBLIC_FIREBASE_APP_ID;
export const FIREBASE_MEASUREMENT =
  process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT;
export const FIREBASE_CLOUD_KEY = process.env.FIREBASE_CLOUD_KEY;
