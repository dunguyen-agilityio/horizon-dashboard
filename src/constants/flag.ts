export const FLAG = {
  'United States': '/usa.webp',
  VietNam: '/vn.webp',
};

export const FLAG_KEYS = Object.keys(FLAG) as (keyof typeof FLAG)[];
