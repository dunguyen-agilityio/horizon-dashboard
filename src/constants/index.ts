export * from './common';
export * from './images';
export * from './messages';
export * from './routes';
export * from './text';
export * from './endpoint';
export * from './session';
