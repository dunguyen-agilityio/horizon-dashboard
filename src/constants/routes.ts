import { Route } from '@/types/route';

export const ENDPOINTS = {
  DASHBOARD: '/dashboard',
  NFT_MARKETPLACE: '/nft-marketplace',
  CONTRIBUTOR: '/contributor',
  KANBAN: '/kanban',
  PROFILE: '/profile',
  SIGN_IN: '/sign-in',
};

export const ROUTES: Route[] = [
  {
    title: 'Main Dashboard',
    href: ENDPOINTS.DASHBOARD,
  },
  { title: 'NFT Marketplace', href: ENDPOINTS.NFT_MARKETPLACE },
  { title: 'Contributor', href: ENDPOINTS.CONTRIBUTOR },
  { title: 'Kanban', href: ENDPOINTS.KANBAN },
  { title: 'Profile', href: ENDPOINTS.PROFILE },
  { title: 'Sign In', href: ENDPOINTS.SIGN_IN },
];

export const FOOTER_ROUTES: Route[] = [
  {
    title: 'Marketplace',
    href: '/nft-marketplace',
  },
  { title: 'License', href: '/license' },
  { title: 'Terms of Use', href: '/terms-of-use' },
  { title: 'Blog', href: '/blog' },
];

export const PUBLIC_ROUTES = {
  FORGET_PASSWORD: '/forget-password',
};
