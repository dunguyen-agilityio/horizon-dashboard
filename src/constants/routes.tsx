import { Route } from '@/types/route';

// Icons
import { Home, Cart, User, Kanban, Chart, Lock } from '@/icons';

export const AUTH_ROUTES = {
  SIGN_IN: '/sign-in',
  SIGN_UP: '/sign-up',
  FORGET_PASSWORD: '/forget-password',
};

export const PRIVATE_ROUTES = {
  KANBAN: '/kanban',
  PROFILE: '/profile',
};

export const PUBLIC_ROUTES = {
  NFT_MARKETPLACE: '/nft-marketplace',
  CONTRIBUTOR: '/contributor',
  LICENSE: '/license',
  BLOG: '/blog',
  TERM_OF_USE: '/terms-of-use',
  DASHBOARD: '/dashboard',
};

export const ROUTES: Route[] = [
  {
    title: 'Main Dashboard',
    href: PUBLIC_ROUTES.DASHBOARD,
    icon: <Home key="homeIcon" />,
  },
  {
    title: 'NFT Marketplace',
    href: PUBLIC_ROUTES.NFT_MARKETPLACE,
    icon: <Cart key="cartIcon" />,
  },
  {
    title: 'Contributor',
    href: PUBLIC_ROUTES.CONTRIBUTOR,
    icon: <Chart key="chartIcon" />,
  },
  {
    title: 'Kanban',
    href: PRIVATE_ROUTES.KANBAN,
    icon: <Kanban key="KanbanIcon" />,
  },
  {
    title: 'Profile',
    href: PRIVATE_ROUTES.PROFILE,
    icon: <User key="userIcon" />,
  },
  {
    title: 'Sign In',
    href: AUTH_ROUTES.SIGN_IN,
    icon: <Lock key="lockIcon" />,
  },
];

export const FOOTER_ROUTES: Omit<Route, 'icon'>[] = [
  {
    title: 'Marketplace',
    href: PUBLIC_ROUTES.NFT_MARKETPLACE,
  },
  { title: 'License', href: PUBLIC_ROUTES.LICENSE },
  { title: 'Terms of Use', href: PUBLIC_ROUTES.TERM_OF_USE },
  { title: 'Blog', href: PUBLIC_ROUTES.BLOG },
];
