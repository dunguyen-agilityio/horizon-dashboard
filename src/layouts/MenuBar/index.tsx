import { useState } from 'react';
import {
  Navbar,
  NavbarContent,
  NavbarMenuToggle,
  NavbarMenu,
  NavbarMenuItem,
} from '@nextui-org/react';

// Components
import { NavItem } from '@/components';
import { usePathname } from 'next/navigation';

// Constants
import { ROUTES, LIST_NAV_ICON } from '@/constants/routes';

// Types
import { Navigate } from '@/types/navigate';

const MenuBar = () => {
  const [isMenuOpen, setIsMenuOpen] = useState(false);

  const pathname = usePathname();

  const listNavigate: Navigate[] = ROUTES.map((route, index) => ({
    ...route,
    icon: LIST_NAV_ICON[index],
  }));

  const handleOpenMenu = () => {
    setIsMenuOpen(!isMenuOpen);
  };

  return (
    <Navbar
      isMenuOpen={isMenuOpen}
      onMenuOpenChange={setIsMenuOpen}
      className="bg-transparent shadow-none"
      classNames={{
        wrapper: 'px-0 bg-transparent h-[1/2]',
      }}
    >
      <NavbarContent className="xl:hidden px-0 flex justify-start dark:bg-indigo-dark">
        <NavbarMenuToggle
          aria-label={isMenuOpen ? 'Close menu' : 'Open menu'}
          className="cursor-pointer xl:hidden p-0 h-[40px] dark:bg-indigo-dark"
        />
      </NavbarContent>

      <NavbarMenu className="mt-8">
        {listNavigate.map(({ title, href, icon }, index) => (
          <NavbarMenuItem key={`${title}-${index}`} onClick={handleOpenMenu}>
            <NavItem
              href={href}
              icon={icon}
              label={title}
              isActive={href === pathname}
            />
          </NavbarMenuItem>
        ))}
      </NavbarMenu>
    </Navbar>
  );
};

export default MenuBar;
