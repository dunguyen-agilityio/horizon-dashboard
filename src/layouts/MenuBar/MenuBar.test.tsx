import { render, screen, fireEvent } from '@testing-library/react';
import MenuBar from '.';
import { ROUTES } from '@/constants/routes';

jest.mock('next/navigation', () => ({
  usePathname: jest.fn().mockImplementation(() => '/'),
}));

describe('MenuBar Component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should match snapshot', () => {
    const { container } = render(<MenuBar />);
    expect(container).toMatchSnapshot();
  });

  it('should render the menu toggle button', () => {
    render(<MenuBar />);
    const menuToggleButton = screen.getByLabelText('Open menu');
    expect(menuToggleButton).toBeInTheDocument();
  });

  it('should toggle menu open and close on link click', () => {
    render(<MenuBar />);

    const toggleButton = screen.getByLabelText('Open menu');
    expect(toggleButton).toBeInTheDocument();

    fireEvent.click(toggleButton);

    const navItem = screen.getByText(ROUTES[0].title);
    expect(navItem).toBeInTheDocument();

    const dashBoardLink = screen.getByRole('link', { name: /dashboard/i });
    fireEvent.click(dashBoardLink);
    expect(dashBoardLink).toBeInTheDocument();

    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });
});
