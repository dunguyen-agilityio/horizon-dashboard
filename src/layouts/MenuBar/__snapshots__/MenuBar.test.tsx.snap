// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar Component Should match snapshot 1`] = `
<div>
  <nav
    class="flex z-40 w-full h-auto items-center justify-center data-[menu-open=true]:border-none sticky top-0 inset-x-0 backdrop-blur-lg data-[menu-open=true]:backdrop-blur-xl backdrop-saturate-150 bg-transparent shadow-none"
    style="--navbar-height: 4rem;"
  >
    <header
      class="z-40 flex gap-4 w-full flex-row relative flex-nowrap items-center justify-between max-w-[1024px] px-0 bg-transparent h-[1/2]"
    >
      <ul
        class="gap-4 h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0 xl:hidden px-0 flex justify-start dark:bg-indigo-dark"
        data-justify="start"
      >
        <button
          aria-label="Open menu"
          aria-pressed="false"
          class="group flex items-center justify-center w-6 rounded-small tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 cursor-pointer xl:hidden p-0 h-[40px] dark:bg-indigo-dark"
          type="button"
        >
          <span
            class="sr-only"
          >
            open navigation menu
          </span>
          <span
            class="w-full h-full pointer-events-none flex flex-col items-center justify-center text-inherit group-data-[pressed=true]:opacity-70 transition-opacity before:content-[''] before:block before:h-px before:w-6 before:bg-current before:transition-transform before:duration-150 before:-translate-y-1 before:rotate-0 group-data-[open=true]:before:translate-y-px group-data-[open=true]:before:rotate-45 after:content-[''] after:block after:h-px after:w-6 after:bg-current after:transition-transform after:duration-150 after:translate-y-1 after:rotate-0 group-data-[open=true]:after:translate-y-0 group-data-[open=true]:after:-rotate-45"
          />
        </button>
      </ul>
    </header>
  </nav>
</div>
`;
